engine/src/cmd/carrier.cpp

```diff
@@ -542,29 +543,36 @@ float Carrier::getCargoVolume(void) const {
     return result;
 }
 
+// TODO: get rid of this helper function and others like it.
+extern std::string getJSONValue(const json::jobject& object, const std::string &key, const std::string &default_value);
+
 Unit *Carrier::makeMasterPartList() {
     unsigned int i;
-    static std::string mpl = vs_config->getVariable("data", "master_part_list", "master_part_list");
-    CSVTable *table = loadCSVTableList(mpl, VSFileSystem::UnknownFile, false);
-
     Unit *ret = new Unit();
     ret->name = "master_part_list";
-    if (table) {
-        vsUMap<std::string, int>::const_iterator it;
-        for (it = table->rows.begin(); it != table->rows.end(); ++it) {
-            CSVRow row(table, it->second);
-            Cargo carg;
-            carg.content = row["file"];
-            carg.category = row["categoryname"];
-            carg.volume = stof(row["volume"], 1);
-            carg.mass = stof(row["mass"], 1);
-            carg.quantity = 1;
-            carg.price = stoi(row["price"], 1);
-            carg.description = row["description"];
-            ret->cargo.push_back(carg);
-        }
-        delete table;
+
+    static std::string json_filename = "master_part_list.json"; //vs_config->getVariable("data", "master_part_list", "master_part_list");
```

Comment:
I'd rather we didn't hard-code the filename if we can help it. But perhaps we can't help it, for this iteration. What do you think, @royfalk ?

Reply:
@stephengtuggy agreed it'd be better not to have it hard-coded especially at this level - it should come in from the command-line parameters and/or configuration file where it can be overridden. As a first pass I'm okay with it - especially since the CSV was hard-coded, but we should at least mark it as something to be improved upon via the settings.

Reply:
OK. Filed an issue, #712 . Approving this PR.

Reply:
I'm not a fan of either of the two options I've used:
1. In the first one, I renamed the csv file provided by some other code to json.
2. In this one, I just hard coded it.

I'll add a TODO here and move on. At some point we'll want an asset manager to give us the files and filenames.

Reply:
Opened an issue instead, as this affects more than just this part of the code.
https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/714